### PR TITLE
Do not keep alive http connection

### PR DESCRIPTION
**Pull request by penghuazhou**
Signed-off-by: penghuazhou <1021556870@qq.com>

_Provide a description of what has been changed_

### Checklist
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs/pull/982
- [x] A PR is opened to update our Helm chart https://github.com/kedacore/charts/pull/335

Relates to #3874

### CODE REVIEW

File: pkg/util/http.go [hunk 1]
```diff
@@ -40,8 +40,10 @@ func CreateHTTPClient(timeout time.Duration, unsafeSsl bool) *http.Client {
 	httpClient := &http.Client{
 		Timeout: timeout,
 		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: unsafeSsl},
-			Proxy:           http.ProxyFromEnvironment,
+			TLSClientConfig:   &tls.Config{InsecureSkipVerify: unsafeSsl},
+			Proxy:             http.ProxyFromEnvironment,
+			DisableKeepAlives: true,
+			IdleConnTimeout:   100 * time.Second,
```
Comment (JorTurFer) [inline pkg/util/http.go:46]: I think we should have the option to configure this. I mean, now always we will want to disable the keep alive, working with some scaler we could be interested on it.
IMO we should add an option to configure this, keeping the old behavior as default

Comment (JorTurFer) [inline pkg/util/http.go:46]: I think we should have the option to configure this. I mean, now always we will want to disable the keep alive, working with some scaler we could be interested on it.
IMO we should add an option to configure this, keeping the old behavior as default

**Review by JorTurFer — COMMENTED** [review]
Thanks for the improvement, I left a comment inline

File: pkg/util/http.go (hunk 1 above)
Comment (penghuazhou) [reply pkg/util/http.go:48]: Now we can configure it by DISABLE-KEEP-ALIVES, keeping the old behavior as default

Comment (JorTurFer) [reply pkg/util/http.go:48]: To reduce the places where we need to change/pass the parameter, what do you think about doing a field in the package populated from the environment variable? 
There are some helpers in the same package, so you could just add another helper for getting bool envs adding this function to `env_resolver.go`:
```go
func ResolveOsEnvBool(envName string, defaultValue bool) (bool, error) {
	valueStr, found := os.LookupEnv(envName)

	if found && valueStr != "" {
		return strconv.ParseBool(valueStr)
	}

	return defaultValue, nil
}
```
and then, use it in `http.go` like:
```go
var disableKeepAlives bool

func init() {
	var err error
	disableKeepAlives, err = ResolveOsEnvBool("KEDA_HTTP_DISABLE_KEEP_ALIVE", false)
	if err != nil {
		disableKeepAlives = false
	}
}
``` 

With this approach, you don't need to pass the value from the main.go until here through all the places.
WRT environment name, please use something like `KEDA_HTTP_XXXXX`. I suggest to use `KEDA_HTTP_DISABLE_KEEP_ALIVE` but if you prefer other name following the pattern, it's okey from my side

**Comment by JorTurFer** [comment]
BTW, in order to be able to merge this, we need some things:

- Please rebase main because there are merge conflicts
- All the commits must be signed, https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

Does it make sense to add this new env variable to helm chart parameters?

File: pkg/util/http.go (hunk 1 above)
Comment (penghuazhou) [reply pkg/util/http.go]: @JorTurFer, I have change it, please review again, thanks.

**Comment by penghuazhou** [comment]
> * Support disable keep http connection alive charts#335

https://github.com/kedacore/charts/pull/335

**Comment by penghuazhou** [comment]
> 

https://github.com/kedacore/charts/pull/335

**Review by zroubalik — APPROVED** [review]

**Comment by zroubalik** [comment]
@tomkerkhove docs are not ready yet https://github.com/kedacore/keda-docs/pull/982

**Comment by JorTurFer** [comment]
This feature was waiting docs and chart

**Pull request merged**